flow/scripts/macro_place.tcl

```diff
@@ -41,14 +48,51 @@ proc find_macros {} {
 
 
 if {[find_macros] != ""} {
-  if {[info exists ::env(MACRO_PLACEMENT)]} {
-    source $::env(SCRIPTS_DIR)/read_macro_placement.tcl
-    puts "\[INFO\]\[FLOW-xxxx\] Using manual macro placement file $::env(MACRO_PLACEMENT)"
-    read_macro_placement $::env(MACRO_PLACEMENT)
+  if {[info exists ::env(RTLMP_FLOW)]} {
+    puts "RTLMP Flow enabled..."
+    partition_design -max_num_inst $env(RTLMP_MAX_INST) \
+                    -min_num_inst $env(RTLMP_MIN_INST) \
+                    -max_num_macro $env(RTLMP_MAX_MACRO) \
+                    -min_num_macro $env(RTLMP_MIN_MACRO) \
+                    -net_threshold 5 -virtual_weight 500 \
+                    -num_hop 5 -timing_weight 1 \
+                    -report_directory $env(RTLMP_RPT_DIR) \
+                    -report_file $env(RTLMP_RPT_FILE)
+ 
+    set flag [rtl_macro_placer -config_file $env(RTLMP_CONFIG_FILE) \
+                                -macro_blockage_file $env(RTLMP_BLOCKAGE_FILE) \
+                                -report_directory $env(RTLMP_RPT_DIR) \
+                                -report_file $env(RTLMP_RPT_FILE)]
+    puts "Delete buffers for RTLMP flow..."
+    remove_buffers
   } else {
-    macro_placement \
-      -halo $::env(MACRO_PLACE_HALO) \
-      -channel $::env(MACRO_PLACE_CHANNEL)
+    if {[info exists ::env(MACRO_PLACEMENT)]} {
```

Comment:
I think this conditional needs to be the other way around. Check if there is a macro placement file first, then check if RTLMP flow is enabled.

If the two options are mutually exclusive, there should be an error if both are set.